### PR TITLE
Update(rbac): remove MCAD, Appwrapper, instascales related permission in operator

### PR DESCRIPTION
**Pull request by zdtsw**
ref: https://issues.redhat.com/browse/RHOAIENG-5129

CFO has removed MCAD and Appwrapper (?) instascales  CRD in their code
we should cleanup such rbac in operator

## Description

a side piece:  golinter failed with reusing old cache to get this PR "green" i update the cache part in GH action. 

## Merge criteria:

- [ ] The commits are squashed in a cohesive manner and have meaningful messages.
- [ ] Testing instructions have been added in the PR body (for PRs involving changes that are not immediately obvious).
- [ ] The developer has manually tested the changes and verified that the changes work

### CODE REVIEW

**Comment by zdtsw** [comment]
to check if this is ok to remove "mcad" and "appwrapper" which might impact pipeline as well @HumairAK

File: controllers/datasciencecluster/kubebuilder_rbac.go [hunk 1]
```diff
@@ -141,12 +141,6 @@ package datasciencecluster
 // +kubebuilder:rbac:groups="monitoring.coreos.com",resources=thanosrulers/status,verbs=get;create;patch;delete;deletecollection
 // +kubebuilder:rbac:groups="monitoring.coreos.com",resources=probes,verbs=get;create;patch;delete;deletecollection
 
-// TODO: cleanup when DSP do the switch
-// +kubebuilder:rbac:groups="mcad.ibm.com",resources=appwrappers,verbs=create;delete;list;patch;get
-
-// +kubebuilder:rbac:groups="workload.codeflare.dev",resources=appwrappers,verbs=create;delete;list;patch;get;deletecollection;update;watch
-// +kubebuilder:rbac:groups="workload.codeflare.dev",resources=appwrappers/finalizers,verbs=create;delete;list;patch;get;deletecollection;update;watch
-// +kubebuilder:rbac:groups="workload.codeflare.dev",resources=appwrappers/status,verbs=create;delete;list;patch;get;deletecollection;update;watch
 // +kubebuilder:rbac:groups="workload.codeflare.dev",resources=queuejobs,verbs=create;delete;list;patch;get;deletecollection;update;watch
```
Comment (sutaakar) [inline controllers/datasciencecluster/kubebuilder_rbac.go:144]: This line can be removed too

**Comment by zdtsw** [comment]
seems the go-linter cache is causing problem for this PR, run it in my forked repo with green result https://github.com/zdtsw-forking/opendatahub-operator/actions/runs/8627717924/job/23648227066

**Review by DharmitD — COMMENTED** [review]
Confirming that removing these resources from RBAC doesn't affect DSP logic.

/lgtm

**Pull request merged**